src/core_plugins/kibana/public/context/components/size_picker/size_picker.js

```diff
@@ -0,0 +1,30 @@
+import _ from 'lodash';
+import uiModules from 'ui/modules';
+import contextSizePickerTemplate from './size_picker.html';
+import './size_picker.less';
+
+const module = uiModules.get('apps/context', [
+  'kibana',
+]);
+
+module.directive('contextSizePicker', function ContextSizePicker() {
+  return {
+    bindToController: true,
+    controller: ContextSizePickerController,
+    controllerAs: 'contextSizePicker',
+    replace: true,
+    restrict: 'E',
+    scope: {
+      count: '=',
+      setCount: '=',
+    },
+    template: contextSizePickerTemplate,
+  };
+});
+
+
+function ContextSizePickerController() {
+  this.getOrSetCount = (count) => (
+    _.isUndefined(count) ? this.count : this.setCount(count)
```

Comment:
> I have added code to properly call $rollbackViewValue when a value change is being propagated from the outside.

The issue I'm concerned about here is the opposite, when the value in the view changes but it might not get propagated to the `size-picker` scope. `ng-model` sets up a two way binding between the value in the input element and the value of `size-picker`'s `$scope.count`. When the view value changes, `setCount` is called and Angular expects this to set the value of `$scope.count`. This might not happen since `setCount` uses a callback that could fail and if it does fail the `size-picker` component will be left in an invalid state.

It's probably not a huge deal in this specific case, but I don't think it's a good pattern to get into the habit of using. Imagine if this were a slightly more complex component. If the view value and the $scope value get out of sync the component controller might operate on old values and show nonsensical results to the user.

> I don't see how two watches would provide any advantage over that.

Since the issue here is that we don't want `size-picker` mutating its parent component's state, I'm suggesting the two watchers as a way to implement a poor man's one-way binding, keeping state mutations local to `size-picker`.